### PR TITLE
Score functions for FData

### LINKED ISSUE #378: Scorers for functional response
Implement scoring methods for functional response (to be used for example in validation and hyperparameter search).

### PR DESCRIPTION
**Pull request by ElenaPetrunina**
Close #378

### CODE REVIEW

**Review by vnmabus — CHANGES_REQUESTED** [review]

File: skfda/misc/__init__.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
     metrics,
     operators,
     regularization,
+    score_functions,
```
[the remark this answers is not in the document]
Comment (ElenaPetrunina) [reply skfda/misc/__init__.py:10]: Rename the file?

[the remark this answers is not in the document]
Comment (vnmabus) [reply skfda/misc/__init__.py:10]: Yes

File: skfda/misc/scoring.py [hunk 2]
```diff
@@ -0,0 +1,1075 @@
...
+            start,
+            end,
+        )
+
+    except ValueError:
+        return float('-inf')
+
+    # If the dimension of the codomain is > 1,
+    # the mean of the scores is taken
+    return np.mean(integral[0]) / (end - start)
```
Comment (vnmabus) [inline skfda/misc/scoring.py:1075]: Probably this should be a function, as it is common between the different scores. Also, why not `_domain_measure` here?

**Review by vnmabus — CHANGES_REQUESTED** [review]

**Comment by vnmabus** [comment]
It appears that the topic of constant predictions has been already discussed in scikit-learn:
https://github.com/scikit-learn/scikit-learn/issues/16857

We should probably try to adapt the same solution as them, either in this PR or in a future fix before the next release: 
https://github.com/scikit-learn/scikit-learn/pull/17266

**Pull request merged**